src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1327,14 +1347,24 @@ private static bool IsNonContextualModifier(SyntaxToken nextToken)
         private bool IsPartialType()
         {
             Debug.Assert(this.CurrentToken.ContextualKind == SyntaxKind.PartialKeyword);
-            switch (this.PeekToken(1).Kind)
+            var nextToken = this.PeekToken(1);
+            switch (nextToken.Kind)
             {
                 case SyntaxKind.StructKeyword:
                 case SyntaxKind.ClassKeyword:
                 case SyntaxKind.InterfaceKeyword:
                     return true;
             }
 
+            if (nextToken.ContextualKind == SyntaxKind.RecordKeyword)
+            {
+                // This is an unusual use of LangVersion. Normally we only produce errors when the langversion
+                // does not support a feature, but in this case we are effectively making a language breaking
+                // change to consider "record" a type declaration in all ambiguous cases. To avoid breaking
+                // older code that is not using C# 9 we conditionally parse based on langversion
+                return IsFeatureEnabled(MessageID.IDS_FeatureRecords);
```

Comment:
>return IsFeatureEnabled(MessageID.IDS_FeatureRecords); [](start = 16, length = 54)

Same comment here.

Reply:
We already took the break in C# 9 that `record` is not a valid type declaration or reference. That's why I'm using the C# 9 LangVer.

---
In reply to: [580449938](https://github.com/dotnet/roslyn/pull/51252#discussion_r580449938) [](ancestors = 580449938)